velox/functions/lib/Re2Functions.cpp

```diff
@@ -767,7 +875,34 @@ std::shared_ptr<exec::VectorFunction> makeLike(
       name,
       inputArgs[1].type->toString());
   auto pattern = constantPattern->as<ConstantVector<StringView>>()->valueAt(0);
-  return std::make_shared<LikeConstantPattern>(pattern, escapeChar);
+  if (!escapeChar) {
+    PatternKind patternKind;
+    vector_size_t reducedLength;
+    std::tie(patternKind, reducedLength) = determinePatternKind(pattern);
```

Comment:
Can this be simplified to 

```
auto [patternKind, reducedLength] = determinePatternKind(pattern);
```

?

Reply:
Thanks for the suggestion, this did not work though so I have retained std::tie here.